src/jit/simd.h

```diff
@@ -5,6 +5,42 @@
 #ifndef _SIMD_H_
 #define _SIMD_H_
 
+// Underlying hardware information
+// This type is used to control
+// 1. The length of System.Numerics.Vector<T>.
+// 2. Codegen of System.Numerics.Vectors.
+// 3. Codgen of floating-point arithmetics (VEX-encoding or not).
+//
+// Note
+// - Hardware SIMD support is classified to the levels. Do not directly use
+//   InstructionSet (instr.h) for System.Numerics.Vectors.
+// - Values of SIMDLevel have strictly increasing order.
+enum SIMDLevel
+{
+    SIMD_Not_Supported = 0,
+#ifdef _TARGET_XARCH_
+    // SSE2 - The min bar of SIMD ISA on x86/x64.
+    // Vector<T> length is 128-bit.
+    // Floating-point instructions are legacy SSE encoded.
+    SIMD_SSE2_Supported = 1,
+
+    // SSE3_4 - RyuJIT generates SSE3, SSSE3, SSE4.1 and SSE4.2 instructions for certain intrinsics.
+    // Vector<T> length is 128-bit.
+    // Floating-point instructions are legacy SSE encoded.
+    SIMD_SSE3_4_Supported = 2,
+
+    // TODO - AVX - Hardware supports AVX instruction set.
+    // TODO - Vector<T> length is 128-bit and SIMD instructions are VEX-128 encoded.
+    // TODO - Floating-point instructions are VEX-128 encoded.
+    SIMD_AVX_Supported = 3,
+
+    // AVX2 - Hardware has AVX and AVX2 instruction set.
+    // Vector<T> length is 256-bit and SIMD instructions are VEX-256 encoded.
+    // Floating-point instructions are VEX-128 encoded.
+    SIMD_AVX2_Supported = 4,
+#endif
```

Comment:
Another question: should this be SIMDLevel and elements, e.g., SIMD_SSE2_Supported, or would it be clearer as SIMDTarget and, e.g., SIMD_Target_SSE2? Doesn't really matter to me.

Reply:
> Doesn't really matter to me.

Me too 😃

Reply:
I don't have a strong opinion, but I'm fine with SIMDLevel and SIMD_xxx_Supported